packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -110,6 +112,8 @@ const fetchDocuments: (
         // $bsonSize is only supported for mongodb >= 4.4.0
         '4.4.0'
       ) &&
+      // ADF doesn't support $bsonSize
+      !isDataLake &&
```

Comment:
Do we also want to have a generic try/catch fallback in case it doesn’t work on some other environments?

Reply:
Sure, is https://github.com/mongodb-js/compass/pull/4614/commits/1dc52ddb130e2a51f374f526917b665c0cf526d1 something that matches what you had in mind?

Reply:
Yes, and still LGTM :+1: